src/smart-components/role/role-resource-definitions-table-helpers.js

```diff
@@ -1,5 +1,14 @@
-export const createRows = (data) =>
-  data.reduce(
+import messages from '../../Messages';
+
+export const createRows = (data, isHosts, intl) => {
+  let finalData = data;
+  if (isHosts) {
+    finalData = data.filter((item) => item !== null);
+    if (finalData.length < data.length) {
+      finalData = [intl.formatMessage(messages.ungroupedSystems), ...finalData];
+    }
```

Comment:
This thing is used for checking if  `data` does not contain `null` value for ungroupped systems, right? I'd personally change it to something like
```
finalData = [...data.includes(null) ? [] : [intl.formatMessage(messages.ungroupedSystems)], ...finalData]
```

Even easier one could be
```
if (isHosts && !data.includes(null)) {
  finalData.unshift(intl.formatMessage(messages.ungroupedSystems))
}
```